models/classes/search/SearchProxy.php

```diff
@@ -100,8 +104,16 @@ public function search(ServerRequestInterface $request): array
         $queryParams = $request->getQueryParams();
         $results = $this->executeSearch($query);
 
-        return $this->getResultSetResponseNormalizer()
+        $response = $this->getResultSetResponseNormalizer()
             ->normalize($query, $results, $queryParams['params']['structure']);
+
+        $response['settings'] = $this->getSearchSettingsService()
+            ->getSettingsByClassMetadataSearchRequest(
+                (new ClassMetadataSearchRequest())
+                    ->setClassUri($request->getQueryParams()['params']['parentNode'])
```

Comment:
nitpick: You already have a local copy of the query params here, the method call can be avoided

```suggestion
                    ->setClassUri($queryParams['params']['parentNode'])
```